secret-provider/src/main/scala/io/lenses/connect/secrets/providers/AWSHelper.scala

```diff
@@ -48,12 +50,26 @@ class AWSHelper(
   private val objectMapper = new ObjectMapper()
 
   // get the key value and ttl in the specified secret
-  override def lookup(secretId: String): Either[Throwable, ValueWithTtl[Map[String, String]]] =
+  override def lookup(secretId: String): Either[Throwable, ValueWithTtl[Map[String, String]]] = {
+    val secretName = getSecretName(secretId)
     for {
-      secretTtl         <- getTTL(secretId)
-      secretValue       <- getSecretValue(secretId)
+      secretTtl         <- getTTL(secretName)
+      secretValue       <- getSecretValue(secretName)
       parsedSecretValue <- parseSecretValue(secretValue)
     } yield ValueWithTtl(secretTtl, parsedSecretValue)
+  }
+
+  private def getSecretName(secretId: String): String = {
+    val hasAccount = secretId.indexOf("$")
```

Comment:
This variable is named as a boolean (hasAccount).  This and the > 1 condition below should be merged so this is a boolean variable.